declarations/Compiler.d.ts

```diff
@@ -0,0 +1,235 @@
+    }>;
+    /** @type {string=} */
+    name: string | undefined;
+    /** @type {Compilation=} */
+    parentCompilation: Compilation | undefined;
+    /** @type {Compiler} */
+    root: Compiler;
+    /** @type {string} */
+    outputPath: string;
+    /** @type {OutputFileSystem} */
+    outputFileSystem: OutputFileSystem;
+    /** @type {IntermediateFileSystem} */
+    intermediateFileSystem: IntermediateFileSystem;
+    /** @type {InputFileSystem} */
+    inputFileSystem: InputFileSystem;
+    watchFileSystem: any;
+    /** @type {string|null} */
+    recordsInputPath: string | null;
+    /** @type {string|null} */
+    recordsOutputPath: string | null;
+    records: {};
+    /** @type {Set<string>} */
+    managedPaths: Set<string>;
+    /** @type {Set<string>} */
+    immutablePaths: Set<string>;
+    /** @type {Set<string>} */
+    removedFiles: Set<string>;
+    /** @type {Map<string, FileSystemInfoEntry | null>} */
+    fileTimestamps: Map<string, FileSystemInfoEntry | null>;
+    /** @type {Map<string, FileSystemInfoEntry | null>} */
+    contextTimestamps: Map<string, FileSystemInfoEntry | null>;
+    /** @type {ResolverFactory} */
+    resolverFactory: ResolverFactory;
+    infrastructureLogger: any;
+    /** @type {WebpackOptions} */
+    options: WebpackOptions;
+    context: string;
+    requestShortener: import("./RequestShortener");
+    cache: import("./Cache");
+    compilerPath: string;
+    /** @type {boolean} */
+    running: boolean;
+    /** @type {boolean} */
+    watchMode: boolean;
+    /** @private @type {WeakMap<Source, { sizeOnlySource: SizeOnlySource, writtenTo: Map<string, number> }>} */
+    _assetEmittingSourceCache: WeakMap<Source, {
+        sizeOnlySource: SizeOnlySource;
+        writtenTo: Map<string, number>;
+    }>;
+    /** @private @type {Map<string, number>} */
+    _assetEmittingWrittenFiles: Map<string, number>;
+    /**
+     * @param {string | (function(): string)} name name of the logger, or function called once to get the logger name
+     * @returns {Logger} a logger with that name
+     */
+    getInfrastructureLogger(name: string | (() => string)): import("./logging/Logger");
+    /**
+     * @param {WatchOptions} watchOptions the watcher's options
+     * @param {Callback<Stats>} handler signals when the call finishes
+     * @returns {Watching} a compiler watcher
+     */
+    watch(watchOptions: import("../declarations/WebpackOptions").WatchOptions, handler: Callback<import("./Stats")>): import("./Watching");
+    /**
+     * @param {Callback<Stats>} callback signals when the call finishes
+     * @returns {void}
+     */
+    run(callback: Callback<import("./Stats")>): void;
+    /**
+     * @param {RunAsChildCallback} callback signals when the call finishes
+     * @returns {void}
+     */
+    runAsChild(callback: RunAsChildCallback): void;
+    purgeInputFileSystem(): void;
+    /**
+     * @param {Compilation} compilation the compilation
```

Comment:
types in these comments should probably dropped or updated to the correct references. `Compilation` is unknown here. It should be `import("./Compilation")`. But as this is a typescript file JSDoc types are incorrect and TSDoc should be used without types. Types are in the signature.